### PR TITLE
Enable gRPC for datastore API.

### PR DESCRIPTION
**Pull request by dhermes**
~~**NOTE**: Has #2097 as diffbase.~~

### CODE REVIEW

**Comment by dhermes** [comment]
@tseaver I don't have a flag for disabling it, but am happy to add it, just wanted to here from you about it.

File: gcloud/datastore/_generated/datastore_grpc_pb2.py [hunk 1]
```diff
@@ -4,6 +4,21 @@
...
+from gcloud.datastore._generated.datastore_pb2 import BeginTransactionResponse
+from gcloud.datastore._generated.datastore_pb2 import CommitRequest
+from gcloud.datastore._generated.datastore_pb2 import CommitResponse
+from gcloud.datastore._generated.datastore_pb2 import LookupRequest
+from gcloud.datastore._generated.datastore_pb2 import LookupResponse
+from gcloud.datastore._generated.datastore_pb2 import RollbackRequest
+from gcloud.datastore._generated.datastore_pb2 import RollbackResponse
+from gcloud.datastore._generated.datastore_pb2 import RunQueryRequest
+from gcloud.datastore._generated.datastore_pb2 import RunQueryResponse
+#   END: Manually added imports
```
Comment (tseaver) [inline gcloud/datastore/_generated/datastore_grpc_pb2.py]: I think I remember wrestling with that issue earlier.  Do we need to fix `scripts/make_operations_grpc.py` to make the manual fudging unnecessary?

File: gcloud/datastore/test_connection.py [hunk 2]
```diff
@@ -14,6 +14,11 @@
 
 import unittest
 
+try:
+    from gcloud.datastore.connection import _HAVE_GRPC
+except ImportError:  # pragma: NO COVER
+    _HAVE_GRPC = False
```
Comment (tseaver) [inline gcloud/datastore/test_connection.py]: The conditional import should be unnecessary:  that name is present whether or not the gRPC stuff can be imported.

**Comment by tseaver** [comment]
The environment flags for pubsub and logging were basically there to allow me to test with the gRPC team, without turning the feature on by default.  We can probably leave them out here, if you don't feel paranoid about the gRPC stuff breaking us.

File: gcloud/datastore/_generated/datastore_grpc_pb2.py (hunk 1 above)
Comment (dhermes) [inline gcloud/datastore/_generated/datastore_grpc_pb2.py]: Maybe? Any ideas how you'd fix the problem?

File: gcloud/datastore/test_connection.py (hunk 2 above)
Comment (dhermes) [inline gcloud/datastore/test_connection.py]: DERP. Good point

**Comment by dhermes** [comment]
> if you don't feel paranoid about the gRPC stuff breaking us

I don't feel paranoid

**Comment by tseaver** [comment]
> I don't feel paranoid

[Hmm](https://github.com/GoogleCloudPlatform/gcloud-python/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3AgRPC%20).

**Comment by dhermes** [comment]
Hmm aside, let's get #2097 settled and then move back here. (I'll resolve the merge conflict.)

**Comment by dhermes** [comment]
@tseaver Rebased after #2097, PTAL.

The question remaining is how to update `make_operations_grpc.py` to have all the imports. Should I just import all the names from `gcloud.datastore._generated.datastore_pb2`?

----

As for worry about gRPC, I was more saying I'm not worried about people having imports work but wanting to turn it off. And those people can just hack and edit the module-private `_HAS_GRPC`

**Comment by tseaver** [comment]
> The question remaining is how to update make_operations_grpc.py to have all the imports. Should I just import all the names from gcloud.datastore._generated.datastore_pb2?

This is the canonical case for `from foo import *`, even.

**Comment by dhermes** [comment]
Trying to avoid the `import *`

**Comment by dhermes** [comment]
@tseaver Check out the latest commit

File: gcloud/datastore/_generated/datastore_grpc_pb2.py [hunk 3]
```diff
@@ -13,11 +7,19 @@
...
 from gcloud.datastore._generated.datastore_pb2 import RollbackResponse
 from gcloud.datastore._generated.datastore_pb2 import RunQueryRequest
 from gcloud.datastore._generated.datastore_pb2 import RunQueryResponse
-#   END: Manually added imports
+#   END: Imports from datastore_pb2
+import grpc
+from grpc.beta import implementations as beta_implementations
+from grpc.beta import interfaces as beta_interfaces
+from grpc.framework.common import cardinality
+from grpc.framework.interfaces.face import utilities as face_utilities
```
Comment (tseaver) [inline gcloud/datastore/_generated/datastore_grpc_pb2.py]: Should imports from a third-party package like `grpc` be in a separate section above the "local" imports?

Comment (tseaver) [inline gcloud/datastore/_generated/datastore_grpc_pb2.py]: Never mind, we're generating the code (as I see below).

**Comment by tseaver** [comment]
LGTM

**Pull request merged**